### PR TITLE
pandas casting issues

### LINKED ISSUE #1726: Behavior of dataarray with no dimensions
Consider

```python
type(np.array([1.0]).mean())
# -> numpy.float64

type(pd.Series([1.0]).mean())
# -> float

type(xr.DataArray([1.0]).mean())
# -> xarray.core.dataarray.DataArray
```

The issue is that this dimensionless data array won't be cast into float by numpy/pandas when constructing a new ndarray/dataframe. You'll have to do it explicitly. Not a big deal but it feels weird.

I'm sure there's a real technical reason (keeping metadata ?) behind this behavior but I couldn't find any discussion about it.

### PR DESCRIPTION
**Pull request by 0x0L**
- [x] Closes #1726 

Added a comment about constructing `pandas` objects from `xarray` objects

### CODE REVIEW

**Comment by fmaussion** [comment]
Thanks for the PR! Doc improvements are always appreciated.

Maybe this would be better placed in the computations section? (http://xarray.pydata.org/en/latest/computation.html)

Also, I don't understand what you mean with ``Constructing `pandas` objects sometimes require explicit casting.`` Maybe we should refer to the working with pandas page then? (http://xarray.pydata.org/en/latest/pandas.html)

**Comment by 0x0L** [comment]
@fmaussion 

In a nutshell,

```python
x = xr.DataArray([1,2,3])
pd.Series({'mean': x.mean()})
```
does not work as expected. You'll want one of

```python
pd.Series({'mean': float(x.mean())})
pd.Series({'mean': x.mean()}, dtype=float)
pd.Series({'mean': x.mean()}).astype(float)
```

**Comment by shoyer** [comment]
Can you merge master? https://github.com/pydata/xarray/pull/1747 should fix the build failure

**Comment by 0x0L** [comment]
@shoyer all good

File: doc/faq.rst [hunk 1]
```diff
@@ -58,6 +58,15 @@ fundamentally multi-dimensional. If your data is unstructured or
 one-dimensional, stick with pandas.
 
 
+Gotchas
```
Comment (shoyer) [inline doc/faq.rst]: Can you frame this as a question, so it looks more similar to the other headings?

e.g., "Why don't aggregations return Python scalars?"

File: doc/faq.rst [hunk 2]
```diff
@@ -58,6 +58,15 @@ fundamentally multi-dimensional. If your data is unstructured or
...
+-------
+
+xarray tries hard to be self-consistent. Operations on xarray objects return
+other xarray objects. In particular, operations like `mean` or `sum` - even
```
Comment (shoyer) [inline doc/faq.rst]: also mention indexing here?

File: doc/faq.rst [hunk 3]
```diff
@@ -58,6 +58,15 @@ fundamentally multi-dimensional. If your data is unstructured or
...
+when applied to all axes - will return xarray objects. Constructing `pandas`
+objects sometimes require explicit casting.
```
Comment (shoyer) [inline doc/faq.rst]: I don't quite understand what this last comment means, maybe an example would help?

File: doc/faq.rst [hunk 4]
```diff
@@ -58,6 +58,15 @@ fundamentally multi-dimensional. If your data is unstructured or
 one-dimensional, stick with pandas.
 
 
+Gotchas
+-------
+
+xarray tries hard to be self-consistent. Operations on xarray objects return
```
Comment (shoyer) [inline doc/faq.rst]: I would note that the self-consistency is that the DataArray methods already return DataArrays (and the same for Dataset), i.e., it preserves the type. This isn't true for NumPy or pandas, which return scalars.

Comment (0x0L) [reply doc/faq.rst]: I just noticed but numpy aggregation funcs return boxed objects that look like an ndarray

Comment (shoyer) [reply doc/faq.rst]: Yes, NumPy and pandas return boxed scalar types (subclasses of `np.generic`).

File: doc/faq.rst (hunk 3 above)
Comment (0x0L) [reply doc/faq.rst]: I'll add something similar to my reply to @fmaussion above

File: doc/faq.rst [hunk 5]
```diff
@@ -58,13 +66,29 @@ fundamentally multi-dimensional. If your data is unstructured or
...
+does not yield the pandas DataFrame we expected. We need to do the type
+conversion ourselves:
+
+.. ipython:: python
 
-xarray tries hard to be self-consistent. Operations on xarray objects return
-other xarray objects. In particular, operations like `mean` or `sum` - even
-when applied to all axes - will return xarray objects. Constructing `pandas`
-objects sometimes require explicit casting.
+    pd.Series({'x': arr[0], 'mean': arr.mean(), 'std': arr.std()}).astype(float)
```
Comment (shoyer) [inline doc/faq.rst]: Can you also note `float(arr[0])` and `arr[0].item()` as two others ways to convert scalars to builtin types?

**Comment by 0x0L** [comment]
@shoyer Thanks for the help for the writing, most of these are your own words ^^

File: doc/whats-new.rst [hunk 6]
```diff
@@ -18,6 +18,11 @@ What's New
 v0.10.1 (unreleased)
 --------------------
 
+Documentation
+~~~~~~~~~~~~~
+
+- New entry `Why don’t aggregations return Python scalars?` in the :doc:`faq`.
```
Comment (shoyer) [inline doc/whats-new.rst]: Consider giving yourself credit here :)

**Comment by jhamman** [comment]
@0x0L - I gave you credit on the whats new page and synced with master. I'll merge once the tests pass here again.

**Comment by 0x0L** [comment]
@jhamman Thank you

**Pull request merged**